hudi-common/src/main/java/org/apache/hudi/common/util/HoodieWriteClientUtils.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.util;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.hudi.common.model.HoodieCommitMetadata;
+import org.apache.hudi.common.table.HoodieTimeline;
+import org.apache.hudi.common.table.timeline.HoodieInstant;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+
+/**
+ * Helper class for utilities required during write time
+ */
+public class HoodieWriteClientUtils {
+
+  private static final Logger LOG = LogManager.getLogger(HoodieWriteClientUtils.class);
+
+  /**
+   * Obtains the average record size based on records written during previous commits. Used for
+   * estimating how many records pack into one file.
+   */
+  public static long averageBytesPerRecord(HoodieTimeline commitTimeline, int defaultRecordSizeEstimate) {
```

Comment:
please move this back to hudi-client. This is just client specific code.. I'd rather leave this as a protected method in HoodieCopyOnWriteTable and have the test be placed under `hudi-client/src/test/org/apache/hudi/table`. Any issues with that?

Reply:
This method was inside a non static inner class. To test this method, unnecessarily had to construct the inner class object. Thats why I pulled it out. Let me put it back into  HoodieCopyOnWriteTable as a protected static method, so it will be testable.